P5/Test/testmeta2010.odd

```diff
@@ -20,176 +20,177 @@
-	 <elementRef key="teiHeader"/>
-	 <elementRef key="text"/>
-	 <elementRef key="title"/>
-	 <elementRef key="titleStmt"/>
+         <elementRef key="TEI"/>
+         <elementRef key="author"/>
+         <elementRef key="back"/>
+         <elementRef key="body"/>
+         <elementRef key="div"/>
+         <elementRef key="fileDesc"/>
+         <elementRef key="publicationStmt"/>
+         <elementRef key="front"/>
+         <elementRef key="head"/>
+         <elementRef key="item"/>
+         <elementRef key="label"/>
+         <elementRef key="list"/>
+         <elementRef key="p"/>
+         <elementRef key="hi"/>
+         <elementRef key="sourceDesc"/>
+         <elementRef key="revisionDesc"/>
+         <elementRef key="change"/>
+         <elementRef key="teiHeader"/>
+         <elementRef key="text"/>
+         <elementRef key="title"/>
+         <elementRef key="titleStmt"/>
 
        </specGrp>
 
        <p>Now make some changes and deletions</p>
        <specGrp xml:id="Changes">
-	 <elementSpec ident="list" mode="change">
-	   <attList>
-	     <attDef ident="type" mode="change">
-	       <constraintSpec mode="add" ident="a1" scheme="schematron">
-		 <constraint>
-		   <sch:assert
-		       test="count(ancestor-or-self::tei:list[1]/tei:item)&lt;2">list
-		 too short </sch:assert></constraint>
-	       </constraintSpec>
-	       <valList type="closed" mode="replace">
-		 <valItem ident="enumerated"/>
-		 <valItem ident="itemized"/>
-	       </valList>
-	     </attDef>
-	   </attList>
-	 </elementSpec>
-	 <classSpec ident="att.canonical" type="atts" mode="delete"/>
-	 <classSpec ident="att.global.rendition" type="atts" mode="change">
-	   <attList>
-	     <attDef ident="rendition" mode="delete"/>
-	   </attList>
-	 </classSpec>
-	 <macroSpec ident="macro.specialPara" mode="change">
-	   <content>
-	     <zeroOrMore xmlns="http://relaxng.org/ns/structure/1.0">
-	       <choice>
-		 <text/>
-		 <ref name="model.phrase"/>
-	       </choice>
-	     </zeroOrMore>
-	   </content>
-	 </macroSpec>
-	 <elementSpec ident="div" mode="change">
-	   <attList>
-	     <attDef ident="type" mode="delete"/>
-	     <attDef ident="rend" mode="change">
-	       <valList type="closed" mode="add">
-		 <valItem ident="i">
-		   <gloss xml:lang="en" versionDate="2014-01-12">italic</gloss>
-		 </valItem>
-		 <valItem ident="b">
-		   <gloss xml:lang="en" versionDate="2014-01-12">bold</gloss>
-		 </valItem>
-	       </valList>
-	     </attDef>
-	   </attList>
-	 </elementSpec>
-	 <classSpec mode="change" type="atts" ident="att.docStatus">
-	   <attList>
-	     <attDef ident="status" mode="change">
-	       <valList  type="closed" mode="replace">
-		 <valItem ident="good"/>
-		 <valItem ident="bad"/>
-	       </valList>
-	     </attDef>
-	   </attList>
-	 </classSpec>
+         <elementSpec ident="list" mode="change">
+           <attList>
+             <attDef ident="type" mode="change">
+               <constraintSpec mode="add" ident="a1" scheme="schematron">
```

Comment:
European subgroup wonders why the constraintSpec is within the attList/attDef and not a child of elementSpec

Reply:
I have answer, but not a _good_ answer.
The difference was that declaring the constraint specification within the attribute definition (as opposed to as a direct child of the element specification) meant that it was fired only for `<list>` elements that actually had a `@type` attribute, regardless of that attribute’s value. That has now been duplicated by using a `@context` of `"tei:list/@type"`.
What I cannot explain is _why_ anyone thought it was import to fire this test only when a `<list>` has a `@type`. Seems nonsensical to me. (Maybe the author wanted to test that it indeed only fired when there was a `@type`?)